lib/features/context-pad/ContextPadProvider.js

```diff
@@ -192,14 +190,26 @@ ContextPadProvider.prototype.getContextPadEntries = function(element) {
       connect = this._connect,
       create = this._create,
       popupMenu = this._popupMenu,
-      rules = this._rules,
       autoPlace = this._autoPlace,
       translate = this._translate,
       appendPreview = this._appendPreview;
 
   var actions = {};
 
   if (element.type === 'label') {
+    if (this._isDeleteAllowed([ element ])) {
+      assign(actions, {
```

Comment:
Could be de-duplicated (cf. delete action later on :arrow_down:).

Reply:
551bba6 extracts adding delete action to a helper function.